server/controllers/profile.js

```diff
@@ -0,0 +1,118 @@
+const Profile = require("../models/Profile");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+exports.getProfile = asyncHandler(async (req, res, next) => {
+  const id = req.body.id;
+
+  const profile = await Profile.find(
+    {
+      user: id,
+    },
+    { user: 0 }
+  );
+
+  if (!profile) {
+    res.status(400);
+    throw new Error("Invalid request");
+  }
+  res.status(200).json({ profile: profile });
+});
+
+exports.getAllProfiles = asyncHandler(async (req, res, next) => {
+  const profiles = await Profile.find({}, { user: 0 });
+
+  if (!profiles) {
```

Comment:
We don't need this check here. If there are no profiles found, the `find` query returns an empty array which can be sent as a response and on the FE we can conditionally display data